### ISSUE TITLE
error: there is no package called 'backports'

### ISSUE THREAD

**Issue by guhjy** [tlverse/delayed#16]
I am using Windows 10, with "backports" already installed:
> devtools::install_github("jeremyrcoyle/delayed")
Downloading GitHub repo jeremyrcoyle/delayed@master
from URL https://api.github.com/repos/jeremyrcoyle/delayed/zipball/master
Installing delayed
"C:/PROGRA~1/R/R-34~1.3/bin/x64/R" --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  "C:/Users/Jinn-Yuh  \
  Guh/AppData/Local/Temp/Rtmpcn73un/devtools9fc422f4a69/jeremyrcoyle-delayed-ceed3a4"  \
  --library="F:/Dropbox/Stat/R/Library" --install-tests 

* installing *source* package 'delayed' ...
** R
** inst
** tests
** preparing package for lazy loading
** help
*** installing help indices
  converting help for package 'delayed'
    finding HTML links ... done
    DelayedClass                            html  
    FutureJob                               html  
    Job                                     html  
    Scheduler                               html  
    SequentialJob                           html  
    bundle_delayed                          html  
    delayed                                 html  
    find_delayed_error                      html  
    make_graph                              html  
    plot.Delayed                            html  
    plot_delayed_shiny                      html  
** building package indices
** installing vignettes
** testing if installed package can be loaded
*** arch - i386
Error: package or namespace load failed for 'delayed':
 .onLoad failed in loadNamespace() for 'checkmate', details:
  call: loadNamespace(name)
  error: there is no package called 'backports'
error: loading failed

*** arch - x64
Error: package or namespace load failed for 'delayed':
 .onLoad failed in loadNamespace() for 'checkmate', details:
  call: loadNamespace(name)
  error: there is no package called 'backports'
ERROR: loading failed for 'i386', 'x64'
* removing 'F:/Dropbox/Stat/R/Library/delayed'
In R CMD INSTALL
Installation failed: Command failed (1)
> sessionInfo()
R version 3.4.3 (2017-11-30)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

Matrix products: default

locale:
[1] LC_COLLATE=Chinese (Traditional)_Taiwan.950 
[2] LC_CTYPE=Chinese (Traditional)_Taiwan.950   
[3] LC_MONETARY=Chinese (Traditional)_Taiwan.950
[4] LC_NUMERIC=C                                
[5] LC_TIME=Chinese (Traditional)_Taiwan.950    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     

other attached packages:
[1] checkmate_1.8.5 backports_1.1.2

loaded via a namespace (and not attached):
 [1] digest_0.6.15        withr_2.1.1         
 [3] R.methodsS3_1.7.1    R6_2.2.2            
 [5] jsonlite_1.5         git2r_0.21.0        
 [7] BiocInstaller_1.14.3 httr_1.3.1          
 [9] curl_3.1             R.oo_1.21.0         
[11] R.utils_2.6.0        devtools_1.13.4     
[13] tools_3.4.3          R.cache_0.13.0      
[15] yaml_2.1.16          compiler_3.4.3      
[17] R.rsp_0.42.0         memoise_1.1.0       
[19] knitr_1.19

**Comment by jeremyrcoyle** [comment]
Seems to be a problem when installing `checkmate` which is a downstream dependency of `delayed`:

```
.onLoad failed in loadNamespace() for 'checkmate', details:
call: loadNamespace(name)
error: there is no package called 'backports'
```

My guess is there's something wrong with your R environment in general, but I don't have a Windows 10 PC handy to test in any case. You might try to take it up with the `checkmate` developers.

**Comment by jeremyrcoyle** [comment]
Possibly caused by an old version of `devtools`: https://github.com/r-lib/devtools/issues/1409

**Comment by JeffreyMaurer** [comment]
I realize I might be necromancing an old thread so I'm sorry, but I encountered this error trying to install d3heatmap in R, because I had assumed the error` Error in hcbHeatmap(lcounts, ColSideColors = cc) : 'ColSideColors' must be a character matrix with ncol(x) rows` referred to that library, a prerequisite to the function I was running, batchQC. 

I tried to install d3heatmap and I ended up here with this backport error. 

The real reason the error occurred was because my count matrix didn't match my batch or condition vectors in length--I needed to get rid of an extra column of metadata.
